OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs

```diff
@@ -0,0 +1,44 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptPropertyGroup("Ability")]
+	public class CaptureProperties : ScriptActorProperties
+	{
+		readonly CapturesInfo normalInfo;
+		readonly ExternalCapturesInfo externalInfo;
+
+		public CaptureProperties(ScriptContext context, Actor self)
+			: base(context, self)
+		{
+			normalInfo = Self.Info.Traits.GetOrDefault<CapturesInfo>();
+			externalInfo = Self.Info.Traits.GetOrDefault<ExternalCapturesInfo>();
+		}
+
+		[Desc("Captures the target actor.")]
+		public void Capture(Actor target)
+		{
+			var normalCapturable = target.Info.Traits.GetOrDefault<CapturableInfo>();
+			var externalCapturable = target.Info.Traits.GetOrDefault<ExternalCapturableInfo>();
+
+			if (normalInfo != null && normalCapturable != null && normalInfo.CaptureTypes.Contains(normalCapturable.Type))
+				Self.QueueActivity(new CaptureActor(Self, target));
+			else if (externalInfo != null && externalCapturable != null && externalInfo.CaptureTypes.Contains(externalCapturable.Type))
+				Self.QueueActivity(new ExternalCaptureActor(Self, Target.FromActor(target)));
```

Comment:
How about an `else throw new InvalidOperationException("Actor X cannot capture")` or something, so when someone screws up it doesn't fail silently.